### ISSUE TITLE
Java Package Lists does not include nested packages

### ISSUE THREAD

**Issue by tkaken** [doxygen/doxygen#10953]
### The Problem
Starting with Doxygen 1.9.4, when running Doxygen optimized for Java, the Package List does not include nested packages. The problem still is happening in 1.11.0.

### Source code used

Here is the code with the nested packages.

```java
// File: MyBase.java
package tryInheritance.nested;

public interface MyBase {
	void doThis();
}
```
```java
//File: MyImpl.java
package tryInheritance.nested;

public class MyImpl implements MyBase {

	@Override
	public void doThis() {

	}
}
```
Here is the code for a class in single level package:

```java
// File: ClassInNonNestedPackage.java
package somePackage;

public class ClassInNonNestedPackage 
{
	public ClassInNonNestedPackage() {}
	public void doThis() {}

}
```
### Screenshots of results

Here are the screen shots of HTML output made using Doxygen 1.9.3 (this works correctly):
![DoxyClassList_1_9_3](https://github.com/doxygen/doxygen/assets/5786782/2d0f37e4-c9f8-471d-91e4-6222cf3e2bc3)
![DoxyPackageList_1_9_3](https://github.com/doxygen/doxygen/assets/5786782/51d92ae4-cc17-4ede-98af-548d5f93f871)

Here are the screenshots made using Doxygen 1.9.4. Note that I got the same results using 1.11.0. (this shows the nested package not in the Package List):
![DoxyClassList_1_9_4](https://github.com/doxygen/doxygen/assets/5786782/8356e4db-2e3f-42c9-8d6d-deef8f1c19a6)
![DoxyPackageList_1_9_4](https://github.com/doxygen/doxygen/assets/5786782/d1ed63f8-40bb-4b8a-8873-6d86d51e62dc)

I  see between the two versions (1.9.3 and 1.9.4), the page title changed from Packages to Package List.

### Extra information
I have attached the zip file [DoxyConfigsAndJavaNestPackageIssueSource.zip](https://github.com/user-attachments/files/15931006/DoxyConfigsAndJavaNestPackageIssueSource.zip) which contains the source code and Doxygen config files I used to generate the output.

**Comment by albert-github** [comment]
Issue was introduced by:
```
Commit: 444dcca03797987cbcd5754d4a6544a0a0436824 [444dcca]
Date: Wednesday, May 4, 2022 7:55:45 PM
Merge pull request #9245 from albert-github/feature/issue_9241

issue #9241 Undocumented external namespaces from .tags files are shown in output Namespace index
```

I've just pushed a proposed patch, pull request #10968

**Comment by albert-github** [comment]
Issue has been fixed differently by means of [269d60d](https://github.com/doxygen/doxygen/commit/269d60d6057f67b587e491ef5b1336fd071b96a8)
(please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.12.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
